pytorch_lightning/trainer/trainer.py

```diff
@@ -1046,10 +1046,16 @@ def _log_hyperparams(self):
 
                 colliding_keys = lightning_hparams.keys() & datamodule_hparams.keys()
                 if colliding_keys:
-                    raise MisconfigurationException(
-                        f"Error while merging hparams: the keys {colliding_keys} are present "
-                        "in both the LightningModule's and LightningDataModule's hparams."
-                    )
+                    inconsistent_keys = []
+                    for key in colliding_keys:
```

Comment:
No need for the variable anymore

```suggestion
                    for key in lightning_hparams.keys() & datamodule_hparams.keys():
```